components/splitter/splitter.component.ts

```diff
@@ -0,0 +1,382 @@
+      </div>
+
+      @if (!last) {
+        <div
+          nz-splitter-bar
+          [ariaNow]="size.percentage * 100"
+          [ariaMin]="size.postPercentMinSize * 100"
+          [ariaMax]="size.postPercentMaxSize * 100"
+          [resizable]="panel.resizable"
+          [collapsible]="panel.collapsible"
+          [active]="movingIndex() === i"
+          [vertical]="nzLayout() === 'vertical'"
+          [lazy]="nzLazy()"
+          [constrainedOffset]="constrainedOffset()"
+          (offsetStart)="startResize(i, $event)"
+          (collapse)="collapse(i, $event)"
+        ></div>
+      }
+    }
+
+    <!-- Fake mask for cursor -->
+    @if (movingIndex() !== null) {
+      <div
+        aria-hidden
+        class="ant-splitter-mask"
+        [class.ant-splitter-mask-horizontal]="nzLayout() === 'horizontal'"
+        [class.ant-splitter-mask-vertical]="nzLayout() !== 'horizontal'"
+      ></div>
+    }
+  `,
+  imports: [NgTemplateOutlet, NzSplitterBarComponent],
+  providers: [NzDestroyService],
+  host: {
+    class: 'ant-splitter',
+    '[class.ant-splitter-horizontal]': 'nzLayout() === "horizontal"',
+    '[class.ant-splitter-vertical]': 'nzLayout() !== "horizontal"'
+  }
+})
+export class NzSplitterComponent {
+  /** ------------------- Props ------------------- */
+  readonly nzLayout = input<NzSplitterLayout>('horizontal');
+  readonly nzLazy = input(false, { transform: booleanAttribute });
+  readonly nzResizeStart = output<number[]>();
+  readonly nzResize = output<number[]>();
+  readonly nzResizeEnd = output<number[]>();
+
+  readonly destroy$ = inject(NzDestroyService);
+  readonly elementRef = inject<ElementRef<HTMLElement>>(ElementRef);
+
+  /** ------------------- Panels ------------------- */
+  // Get all panels from content children
+  readonly panels = contentChildren(NzSplitterPanelComponent);
+  // Subscribe to the change of properties
+  readonly panelProps = computed(() =>
+    this.panels().map(panel => ({
+      defaultSize: panel.nzDefaultSize(),
+      size: panel.nzSize(),
+      min: panel.nzMin(),
+      max: panel.nzMax(),
+      resizable: panel.nzResizable(),
+      collapsible: coerceCollapsible(panel.nzCollapsible()),
+      contentTemplate: panel.contentTemplate()
+    }))
+  );
+
+  /** ------------------- Sizes ------------------- */
+  /**
+   * The size of the container, used to calculate the percentage size and flex basis of each panel.
+   */
+  readonly containerSize = computed(() =>
+    this.nzLayout() === 'horizontal'
+      ? this.elementRef.nativeElement.clientWidth || 0
+      : this.elementRef.nativeElement.clientHeight || 0
+  );
+  readonly innerSizes = linkedSignal({
+    source: this.panelProps,
+    computation: source => source.map(panel => panel.defaultSize)
+  });
+  readonly sizes = computed(() => {
+    let emptyCount = 0;
+    const containerSize = this.containerSize();
+    const innerSizes = this.innerSizes();
+    const sizes = this.panelProps().map((panel, index) => {
```

Comment:
Can the lambda function be extracted with a good name? It would help in readability imo.